servers/src/main/java/tachyon/worker/block/evictor/LRFUEvictor.java

```diff
@@ -0,0 +1,315 @@
+      // nowhere to transfer blocks to, so evict them
+      plan.toEvict().addAll(candidateBlocks);
+    } else {
+      BlockStoreLocation dest = candidateNextDir.toBlockStoreLocation();
+      for (long block : candidateBlocks) {
+        plan.toMove().add(new Pair<Long, BlockStoreLocation>(block, dest));
+      }
+    }
+    return candidateDirView;
+  }
+
+  @Override
+  public EvictionPlan freeSpaceWithView(long bytesToBeAvailable, BlockStoreLocation location,
+      BlockMetadataManagerView view) {
+    synchronized (mBlockIdToLastUpdateTime) {
+      updateCRFValue();
+      mManagerView = view;
+
+      List<Map.Entry<Long, Double>> sortedCRF = getSortedCRF();
+      List<Pair<Long, BlockStoreLocation>> toMove =
+          new ArrayList<Pair<Long, BlockStoreLocation>>();
+      List<Long> toEvict = new ArrayList<Long>();
+      EvictionPlan plan = new EvictionPlan(toMove, toEvict);
+      StorageDirView candidateDir = cascadingEvict(bytesToBeAvailable, location, plan, sortedCRF);
+
+      if (candidateDir == null) {
+        return null;
+      }
+
+      return plan;
+    }
+  }
+  
+  /**
+   * Sort all blocks in ascending order of CRF
+   * 
+   * @return the sorted CRF of all blocks
+   */
+  private List<Map.Entry<Long, Double>> getSortedCRF() {
+    List<Map.Entry<Long, Double>> sortedCRF = 
+        new ArrayList<Map.Entry<Long, Double>>(mBlockIdToCRFValue.entrySet());
+    Collections.sort(sortedCRF, new Comparator<Map.Entry<Long, Double>>() {
+      public int compare(Entry<Long, Double> o1, Entry<Long, Double> o2) {
+        double res = o1.getValue() - o2.getValue();
+        if (res < 0) {
+          return -1;
+        } else if (res > 0) {
+          return 1;
+        } else {
+          return 0;
+        }
+      }
+    });
+    return sortedCRF;
+  }
+
+  /**
+   * Calculate function F(t) = pow (1.0 / {@link #mAttenuationFactor}, t * {@link #mStepFactor})
+   * 
+   * @param logicTimeInterval time interval since that access to current
+   * @return Function value of F(t)
+   */
+  private double calculateFunction(long logicTimeInterval) {
+    return Math.pow(1.0 / mAttenuationFactor, logicTimeInterval * mStepFactor);
+  }
+
+  /**
+   * This function is used to update CRF of all the blocks according to current logic time. When
+   * some block is accessed in some time, only CRF of that block itself will be updated to current
+   * time, other blocks who are not accessed recently will only be updated until 
+   * {@link #freeSpaceWithView(long, BlockStoreLocation, BlockMetadataManagerView)} is called 
+   * because blocks need to be sorted in the increasing order of CRF. When this function is called,
+   * {@link #mBlockIdToLastUpdateTime} and {@link #mBlockIdToCRFValue} need to be locked in case
+   * of the changing of values.
+   */
+  private void updateCRFValue() {
+    long currentLogicTime = mLogicTimeCount.get();
+    for (Iterator<Map.Entry<Long, Double>> it = mBlockIdToCRFValue.entrySet().iterator(); it
```

Comment:
Can you use something like `for (Entry<Long,Double> entry : mBlockIdToCRFValue.entrySet())` here?